### DISCUSSION TITLE
New Pattern for linking to logs

### DISCUSSION THREAD

**Discussion by johannaweintraub** [twilio-labs/paste#1161] [Q&A] [answered]
Now that we have a dedicated monitoring area and to reduce redundant content across console, I'm working on a new pattern for linking to logs from an object. For example, when viewing a phone number, I should be able to link into the monitor tab with a filtered view of messaging or call logs. Same thing should be possible for any object (a service, function, Trunk, SIM, etc.). I have one concept where the link is next to the table row (for table views) and next to the H1 (for detail views). Most likely, we'd open the filtered logs in a separate tab. You can view prototype below. I'd like your feedback on the use of these components and the placement of the links. I want to make sure this pattern won't conflict with any other planned patterns in the queue. If you have other ideas for a universal placement in the detail page, let me know. I will do some testing on this later this month. 

<img width="957" alt="Screen Shot 2021-02-03 at 3 48 23 PM" src="https://user-images.githubusercontent.com/6243878/106824235-457d3500-6637-11eb-8375-8144c4a3995e.png">
<img width="975" alt="Screen Shot 2021-02-03 at 3 48 17 PM" src="https://user-images.githubusercontent.com/6243878/106824239-4746f880-6637-11eb-93e4-73d9fd9ae8e4.png">

https://www.figma.com/proto/PSU37JsxvOKUFqbfN6mQir/Deep-linking-to-monitor?node-id=9%3A2194&scaling=min-zoom

**Comment by kate-porter** [comment]
Hey Jo! Thanks for looping us into this. I'll spend a bit more time reviewing the designs and sharing detailed feedback.

At first glance, it feels like the "view logs" action in the table row could be built into the `Data grid/table actions` pattern that's in the backlog. In that pattern, we were thinking we'd probably have table-level actions (like select all, row sorting, etc.) and row-level actions. What do you think about contributing row-level actions to that pattern?

I'm not certain if the "view logs" action at the page-level fits into any of the patterns we have in the backlog...we'll just want to make sure placement of the "view logs" action jives well with the `Edit object` pattern which Ishita is working on currently.

**Comment by johannaweintraub** [comment]
That's a great idea about the built in table functionality. Thanks for taking a look! I probably won't be able to take on any add'l patterns until at least Q2. I have several that I'm already working on this quarter.

**Comment by kate-porter** [comment, marked as answer]
Hey @johannaweintraub - I just took a deeper dive into the designs, and no real concerns on my end! Component usage looks good to me. I dropped in a couple of thoughts but they weren't really specific to your component usage/accessibility/etc. Overall, looks great.

As you work on finalizing the placement of the View Logs action on an object detail page, I would just ask that you try to sync with Ishita on her work on the Edit Object pattern. She's just getting started on it, so you might be a bit ahead of her, but it'd just be good to make sure the View Logs action will jive well with any other actions related to editing an object.

**Comment by johannaweintraub** [comment]
Thanks so much Katie! I really appreciate your time and feedback. I'll sync w/ Ishita for sure. I'm going to explore and test a few different directions for this but wanted to be sure this didn't conflict with any planned patterns.
